spec.html

```diff
@@ -34179,30 +34182,30 @@ <h1>String.prototype.padStart ( _maxLength_ [ , _fillString_ ] )</h1>
         <p>This method performs the following steps when called:</p>
         <emu-alg>
           1. Let _O_ be ? RequireObjectCoercible(*this* value).
-          1. Return ? StringPad(_O_, _maxLength_, _fillString_, ~start~).
+          1. Let _S_ be ? ToString(_O_).
+          1. Let _intMaxLength_ be ℝ(? ToLength(_maxLength_)).
+          1. If _fillString_ is not *undefined*, set _fillString_ to ? ToString(_fillString_).
+          1. Return StringPad(_S_, _intMaxLength_, _fillString_, ~start~).
         </emu-alg>
 
         <emu-clause id="sec-stringpad" type="abstract operation">
           <h1>
             StringPad (
-              _O_: an ECMAScript language value,
-              _maxLength_: an ECMAScript language value,
-              _fillString_: an ECMAScript language value,
+              _S_: a String,
+              _maxLength_: a non-negative integer,
+              _fillString_: a String or *undefined*,
               _placement_: ~start~ or ~end~,
-            ): either a normal completion containing a String or a throw completion
+            ): a String
           </h1>
           <dl class="header">
           </dl>
           <emu-alg>
-            1. Let _S_ be ? ToString(_O_).
-            1. Let _intMaxLength_ be ℝ(? ToLength(_maxLength_)).
             1. Let _stringLength_ be the length of _S_.
-            1. If _intMaxLength_ ≤ _stringLength_, return _S_.
```

Comment:
Actually, I just noticed that this early exit would skip the effect of `ToString(_fillString_)` in some cases. To keep this editorial, we'll have to copy this early exit into `padStart`/`padEnd`. That's unfortunate.

Reply:
This seems to be adding lots of extra steps at the callsites - why is "the AO is infallible" a valuable enough benefit to outweigh the readability hit being taken here?

Reply:
That is really unfortunate, I've updated the PR to include the early exit into `padStart`/`padEnd`. If this issue needs further work, I'd happily update the PR as you see fit.